### PR TITLE
New dataset added

### PR DESCRIPTION
**Pull request by Volodimirich**

### CODE REVIEW

File: src/modules/data.py [hunk 1]
```diff
@@ -34,18 +34,34 @@ def get_data(dataset_dict: dict):
...
+        labels = tokenizer(text_arrays, max_length=target_max_length, truncation=True)
+        model_inputs["labels"] = labels["input_ids"]
+        model_inputs['pos_label'] = text_labels
+    elif mode == 'made_data':
+        model_inputs = tokenizer(text=examples['query'],
+                                max_length=input_max_length,
+                                truncation=True, padding="max_length")
+        labels = tokenizer(examples['body'], max_length=target_max_length, truncation=True)
+        model_inputs["labels"] = labels["input_ids"]
+        model_inputs['pos_label'] = examples['labels']
```
Comment (betepok506) [inline src/modules/data.py]: В чем разница этих двух случаев?

File: src/modules/data.py [hunk 2]
```diff
@@ -83,8 +99,34 @@ def groups_texts(examples, tokenizer, block_size):
...
+                                   return_length=True,
+                                   )["input_ids"] for k in
+                      concatenated_text.keys()}
+
+    result = {
+        "input_ids": []
+    }
+    for k, t in tokenized_text.items():
+        result["input_ids"].extend(
+            [t[i: i + block_size] for i in range(0, len(t) - block_size, block_size)])
```
Comment (betepok506) [inline src/modules/data.py]: При такой реализации разве будет работать пайплайн оценки, где нужны labels?

File: src/modules/data.py [hunk 3]
```diff
@@ -94,19 +136,23 @@ def get_dataset(dataset_dict, path_list, tokenizer, total_samples=500,
...
-                                                   'validation': [val_path]})
+        
+        if dataset_name == 'part_data':
+            dataset = load_dataset('json', data_files={'train': [train_path],
+                                                    'test': [test_path],
+                                                    'validation': [val_path]})
+        else:
+            dataset = load_dataset('csv', data_files={'train': [train_path]})
 
         train_dataset = dataset['train'].select(range(total_samples))
```
Comment (betepok506) [inline src/modules/data.py]: Давай вот тут ограничим чтобы не гадать с размером данных dataset['train'].select(range(min(total_samples, len(dataset["train"]))))

File: src/modules/data.py [hunk 4]
```diff
@@ -34,18 +34,34 @@ def get_data(dataset_dict: dict):
         path_dict = {'train': os.path.join(data_path, 'train.jsonl'),
                      'test': os.path.join(data_path, 'test.jsonl'),
                      'val': os.path.join(data_path, 'val.jsonl')}
+    elif dataset_name == 'made_data':
+        path_dict = {'train': os.path.join(data_path, 'filtred_df.csv'),
```
Comment (betepok506) [inline src/modules/data.py:39]: filtered_df.csv

**Review by betepok506 — COMMENTED** [review]
Еще один момент, в train.py для GPT нужен другой data_collator, у меня это в ветке есть. 
 data_collator = DataCollatorForLanguageModeling(tokenizer=tokenizer, mlm=params.dataset.mlm)

File: src/modules/data.py (hunk 1 above)
Comment (Volodimirich) [reply src/modules/data.py]: Вложенность данных, код идентичен. В датасете с hugging face там в поле passage_text лежит и сам текст и метки.

File: src/modules/data.py (hunk 2 above)
Comment (Volodimirich) [reply src/modules/data.py]: Изучу данный момент, забыл про оценку

File: src/modules/data.py (hunk 3 above)
Comment (Volodimirich) [reply src/modules/data.py]: Принято, починю

File: src/modules/data.py (hunk 4 above)
Comment (Volodimirich) [reply src/modules/data.py:39]: Sad noises

File: src/generating_model_responses.py [hunk 5]
```diff
@@ -0,0 +1,88 @@
...
+
+    logger.info(f"Currently used device: {device}")
+
+    logger.info(f"The folder where the result is saved: {folder_result_name}")
+
+    model_info = f'Pretrained {params.model.model_name}' if not params.model.use_local \
+        else f'Local {params.model.model_name} from {params.model.local_path}'
+    logger.info(f'Initializing the model: {model_info}')
+
+    tokenizer = get_tokenizer(params.model.tokenizer_name)
```
Comment (Maksimov-Dmitry) [inline src/generating_model_responses.py:45]: Я думаю на predict лучше доставать tokenizer тоже из checkpoint, чтобы потом не делать приседаний с resize

**Review by Maksimov-Dmitry — APPROVED** [review]
Я бы tokenizer из чекпоинта доставал где мы делаем, в остальном отлично, для t5 кое-что придется изменить(к примеру optimizer), но думаю сделаем это в отдельной ветке
